core-api/src/worker.rs

```diff
@@ -491,3 +502,70 @@ impl PollerBehavior {
         Ok(())
     }
 }
+
+#[derive(Clone, Debug)]
+pub enum WorkerVersioningStrategy {
+    /// Don't enable any versioning
+    None {
+        /// Build ID may still be passed as a way to identify the worker, or may be left empty.
+        build_id: String,
+    },
+    /// Maybe use the modern deployment-based versioning, or just pass a deployment version.
+    WorkerDeploymentBased(WorkerDeploymentOptions),
+    /// Use the legacy build-id-based whole worker versioning.
+    LegacyBuildIdBased {
+        /// A Build ID to use, must be non-empty.
+        build_id: String,
+    },
```

Comment:
Would support a future where this option is deleted when versioning v2 is deleted